packages/contracts-bedrock/test/properties/medusa/ProtocolAtomic.t.sol

```diff
@@ -0,0 +1,151 @@
+import { SafeCall } from "src/libraries/SafeCall.sol";
+import { FuzzERC20 } from "../helpers/Utils.sol";
+
+contract MockCrossDomainMessenger {
+    address public crossDomainMessageSender;
+    address public crossDomainMessageSource;
+    mapping(uint256 chainId => mapping(bytes32 reayDeployData => address)) internal superTokenAddresses;
+    mapping(address => bytes32) internal superTokenInitDeploySalts;
+    // test-specific functions
+
+    function crossChainMessageReceiver(address sender, uint256 destinationChainId) external returns (OptimismSuperchainERC20) {
+        return OptimismSuperchainERC20(superTokenAddresses[destinationChainId][superTokenInitDeploySalts[sender]]);
+    }
+
+    function registerSupertoken(bytes32 deploySalt, uint256 chainId, address token) external {
+        superTokenAddresses[chainId][deploySalt] = token;
+        superTokenInitDeploySalts[token] = deploySalt;
+    }
+    // mocked functions
+
+    function sendMessage(uint256 chainId, address recipient, bytes memory message) external returns (address) {
+        address crossChainRecipient = superTokenAddresses[chainId][superTokenInitDeploySalts[msg.sender]];
+        if (crossChainRecipient == msg.sender) {
+            require(false, "same chain");
+        }
+        crossDomainMessageSender = crossChainRecipient;
+        crossDomainMessageSource = msg.sender;
+        SafeCall.call(crossDomainMessageSender, 0, message);
+        crossDomainMessageSender = address(0);
+    }
+}
+
+contract ProtocolAtomicFuzz is Test {
+    uint8 internal constant MAX_CHAINS = 4;
+    address internal constant BRIDGE = Predeploys.L2_STANDARD_BRIDGE;
+    MockCrossDomainMessenger internal constant MESSENGER =
+        MockCrossDomainMessenger(Predeploys.L2_TO_L2_CROSS_DOMAIN_MESSENGER);
+    OptimismSuperchainERC20 internal superchainERC20Impl;
+    string[] internal WORDS = ["FANCY", "TOKENS"];
+    uint8[] internal DECIMALS = [0, 6, 18, 36];
+
+    struct TokenDeployParams {
+        uint8 remoteTokenIndex;
+        uint8 name;
+        uint8 symbol;
+        uint8 decimals;
+    }
+
+    address[] internal remoteTokens;
+    address[] internal allSuperTokens;
+    mapping(bytes32 => uint256) internal superTokenTotalSupply;
+    mapping(bytes32 => uint256) internal superTokensTotalSupply;
+
+    constructor() {
+        vm.etch(address(MESSENGER), address(new MockCrossDomainMessenger()).code);
+        superchainERC20Impl = new OptimismSuperchainERC20();
+    }
+
+    modifier validateTokenDeployParams(TokenDeployParams memory params) {
+        params.remoteTokenIndex = uint8(bound(params.remoteTokenIndex, 0, remoteTokens.length - 1));
+        params.name = uint8(bound(params.name, 0, WORDS.length - 1));
+        params.symbol = uint8(bound(params.symbol, 0, WORDS.length - 1));
+        params.decimals = uint8(bound(params.decimals, 0, DECIMALS.length - 1));
+        _;
+    }
+
+    function fuzz_DeployNewSupertoken(
```

Comment:
Missing assertions?

Reply:
I can think of an assertion for property 14, but am I missing another one?

Reply:
I mean is ok if is not any property and you're only testing that the setup indeed works. But don't you think using a `fuzz_setup` with a comment explaining this makes it more clear?